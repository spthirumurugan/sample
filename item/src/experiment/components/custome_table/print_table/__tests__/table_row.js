import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import TableRow, { getCellValues } from '../table_row';
import dummyState from '../../__mock__/dummy_state';

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('components TableRow', () => {
  describe('<TableRow />', () => {
    it('should render', () => {
      const store = mockStore(dummyState);
      const comp = renderer.create(
        <TableRow store={store} datasetId={'data_1'} viewId={'view_1'} />
      ).toJSON();
      expect(comp).toMatchSnapshot();
    });
  });
  describe('formatProps func', () => {
    it(`should read the view.filter.props and return an array
      of props from the dataset to display`, () => {
      const { data_1, view_1 } = dummyState.notes.byId;
      const actual = getCellValues(data_1, view_1);
      const expected = ['1'];
      expect(actual).toEqual(expected);
    });
  });
});
