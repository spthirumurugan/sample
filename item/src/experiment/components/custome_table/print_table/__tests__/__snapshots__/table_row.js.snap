// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components TableRow <TableRow /> should render 1`] = `
<tr>
  <td>
    1
  </td>
</tr>
`;
